cmd/scality-cosi-driver/main.go

```diff
@@ -50,5 +50,6 @@ func main() {
 	// Call the run function (defined in cmd.go)
```

Comment:
I can't write a comment above this line, but at `main.go:42`: doesn't `ctx.Done()` close immediately when `cancel` is called? I don't think checking that is worth it. The timeout is good though. We could however add another read on the `sigs` channel in the select if we wanted to force shutdown on multiple `SIGINT`.

Reply:
you are right the select block is indeed redundant.
When cancel() is called, ctx.Done() immediately closes, and the select statement in the goroutine will always choose the <-ctx.Done() case first. This makes the select block somewhat redundant in its current form.

Reply:
I am thinking to change it to something like this
What do you think?

```
go func() {
    sig := <-sigs
    klog.InfoS("Signal received", "type", sig)
    cancel()

    klog.InfoS("Scality COSI driver shutdown initiated successfully, context canceled")

    select {
    case sig = <-sigs:
        klog.ErrorS(nil, "Force shutdown due to repeated signal", "type", sig)
        os.Exit(1)
    case <-time.After(shutdownTimeout):
        klog.ErrorS(nil, "Force shutdown due to timeout", "timeout", shutdownTimeout)
        os.Exit(1)
    }
}()
```

Reply:
I think this is good :+1: 

Perhaps the first log could say something like "Initiating graceful shutdown, repeat signal to force shutdown"?

Reply:
Okay I will create another PR for this, its out of scope for this one. Thanks